### PR TITLE
Hotelreservation

### PR DESCRIPTION
**Pull request by subpal**
Added Hotel Reservation Project

### CODE REVIEW

File: HotelReservation/src/main/java/Day.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+enum Day{
+    WEEKEND(1), WEEKDAY(2);
+    private int id;
+
+    Day(int value){
```
Comment (Himanshu-it) [inline HotelReservation/src/main/java/Day.java]: Why have you added id to Day? Enums are just  constants. Further there does not seem any requirement of adding a private variable id.

File: HotelReservation/src/test/java/HotelTests.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class HotelTests {
+    private static HotelList myHotels = new HotelList();
```
Comment (Himanshu-it) [inline HotelReservation/src/test/java/HotelTests.java:13]: make these variables private. Even make setDefaultsHotel() public without static

File: HotelReservation/src/main/java/Day.java (hunk 1 above)
Comment (subpal) [reply HotelReservation/src/main/java/Day.java]: We need a integer value for each enum, it is used for overriding hashcode function in Request class.

**Review by Himanshu-it — COMMENTED** [review]
Try to group java files in different packages. for example Customer, Days etc can go to Models, and HotelFinder in a service package.
You can think of some diff packages also

File: HotelReservation/src/main/java/Day.java (hunk 1 above)
Comment (Himanshu-it) [reply HotelReservation/src/main/java/Day.java]: Okay. But in that case it is better to write a wrapper class which contains an id and a type as Day
and make that class as the key of hashmap.

File: HotelReservation/src/main/java/TypeOfCustomer.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+enum Type{
```
Comment (shgupta22) [inline HotelReservation/src/main/java/TypeOfCustomer.java]: Rename it to CustomerType. its misleading as Type could be anything

File: HotelReservation/src/main/java/Request.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import com.sun.corba.se.impl.encoding.TypeCodeOutputStream;
+
+public class Request {
```
Comment (shgupta22) [inline HotelReservation/src/main/java/Request.java]: You have added lambok dependency in pom but you are not using that. By using that you don't need to write getter setter equal and hashcode function by your own

File: HotelReservation/src/main/java/HotelList.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    }
+    public void addHotel(Hotel h){
+        listOfHotels.add(h);
+    }
+
+    public void removeHotel(Hotel h){
+        listOfHotels.remove(h);
+    }
+
+    public int size(){
```
Comment (shgupta22) [inline HotelReservation/src/main/java/HotelList.java]: change this api to get the count of hotels, size is not clarifying what size it is

File: HotelReservation/src/main/java/HotelList.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        return listOfHotels.size();
+    }
+
+    public Boolean contains(String h){
+        for(int i = 0; i < listOfHotels.size(); ++i){
```
Comment (shgupta22) [inline HotelReservation/src/main/java/HotelList.java]: you don't need to explicitly iterate over the list. use api provided by java https://www.tutorialspoint.com/java/util/arraylist_contains.htm

File: HotelReservation/src/main/java/HotelList.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+            if(listOfHotels.get(i).getHotelName().equals(h)){
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public Hotel getHotel(String name){
+        for(int i = 0; i < listOfHotels.size(); ++i){
+            if(listOfHotels.get(i).getHotelName().equals(name)){
```
Comment (shgupta22) [inline HotelReservation/src/main/java/HotelList.java]: same as above https://www.tutorialspoint.com/java/util/arraylist_contains.htm

File: HotelReservation/src/main/java/HotelFinder.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
+import java.util.*;
+
+class SortArgument {
```
Comment (shgupta22) [inline HotelReservation/src/main/java/HotelFinder.java]: Don't write multiple classes in a single file. Create different files for each class

File: HotelReservation/src/main/java/Hotel.java [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import java.util.HashMap;
+
+public class Hotel {
```
Comment (shgupta22) [inline HotelReservation/src/main/java/Hotel.java]: use lambok for boilerplate code

**Review by shgupta22 — CHANGES_REQUESTED** [review]
There are lots of changes required, we will discuss them later. Apart from that, you need to change a lot of things. Please work on those.

File: HotelReservation/src/main/java/HotelList.java (hunk 6 above)
Comment (subpal) [reply HotelReservation/src/main/java/HotelList.java]: This function just takes a string `h` as hotelname but, the arraylist contains hotel objects which are not comparable with string `h`.

**Pull request merged**